armi/reactor/tests/test_blocks.py

```diff
@@ -1454,33 +1454,33 @@ def test_106_getAreaFractions(self):
     def test_rotatePins(self):
         b = self.block
         b.setRotationNum(0)
-        index = b.rotatePins(0, justCompute=True)
+        index = b._rotatePins(0, justCompute=True)
         self.assertEqual(b.getRotationNum(), 0)
         self.assertEqual(index[5], 5)
         self.assertEqual(index[2], 2)  # pin 1 is center and never rotates.
 
-        index = b.rotatePins(1)
+        index = b._rotatePins(1)
         self.assertEqual(b.getRotationNum(), 1)
         self.assertEqual(index[2], 3)
         self.assertEqual(b.p.pinLocation[1], 3)
 
-        index = b.rotatePins(1)
+        index = b._rotatePins(1)
         self.assertEqual(b.getRotationNum(), 2)
         self.assertEqual(index[2], 4)
         self.assertEqual(b.p.pinLocation[1], 4)
 
-        index = b.rotatePins(2)
-        index = b.rotatePins(4)  # over-rotate to check modulus
+        index = b._rotatePins(2)
+        index = b._rotatePins(4)  # over-rotate to check modulus
         self.assertEqual(b.getRotationNum(), 2)
         self.assertEqual(index[2], 4)
         self.assertEqual(index[6], 2)
         self.assertEqual(b.p.pinLocation[1], 4)
         self.assertEqual(b.p.pinLocation[5], 2)
 
-        self.assertRaises(ValueError, b.rotatePins, -1)
-        self.assertRaises(ValueError, b.rotatePins, 10)
-        self.assertRaises((ValueError, TypeError), b.rotatePins, None)
-        self.assertRaises((ValueError, TypeError), b.rotatePins, "a")
+        self.assertRaises(ValueError, b._rotatePins, -1)
+        self.assertRaises(ValueError, b._rotatePins, 10)
+        self.assertRaises((ValueError, TypeError), b._rotatePins, None)
+        self.assertRaises((ValueError, TypeError), b._rotatePins, "a")
```

Comment:
Can we get a unit test showing that the new `rotate()` and `_rotatePins()` work for blocks/assemblies that don't have pins at all?  I was looking to figure it out and then realized it would be an informative test to have.

Looking at your code, that would have to still be a `HexBlock`, right?

Reply:
`Block.rotate` is not implemented, only `HexBlock.rotate`

https://github.com/terrapower/armi/blob/8feea21034480954bc793ffd8901ffee36c96b3f/armi/reactor/blocks.py#L1538-L1546

Similarly, there is no `Block.rotatePins`, it's only defined on `HexBlock`.

Are you looking for some tests on an empty `HexBlock` with no children? So it doesn't have any pins?

Reply:
Yeah, I just want a test that shows (now and 3 years from now, when people have monkied with the feature a lot) that the rotation still works in the case there are no pins.

Or show that it fails in the way you want it to.